cli/format.go

```diff
@@ -384,3 +418,56 @@ func applyFormatters(ctx context.Context) func() error {
 		return nil
 	}
 }
+
+func findUp(searchDir string, fileName string) (path string, dir string, err error) {
+	for _, dir := range eachDir(searchDir) {
+		path := filepath.Join(dir, fileName)
+		if fileExists(path) {
+			return path, dir, nil
```

Comment:
how does this handle symlinks? Would it follow a directory symlink that might "escape" into a completely different directory tree?

Reply:
eachDir is only decomposing the /path/ string, removing one component after the other. I don't like to follow symlinks because it can lead to infinite loops and other weirdnesses.

fileExists will return true if the file is readable. It can be a symlink to another file, that's fine.